spices/SPICE-0011-const-object-members.adoc

```diff
@@ -0,0 +1,111 @@
+= Const Object Members
+
+* Proposal: link:./SPICE-0011-const-object-members.adoc[SPICE-0011]
+* Author: https://github.com/bioball[Dan Chao]
+* Status: Review
+* Implemented in: TBD
+* Category: Language
+
+== Introduction
+
+This SPICE proposes allowing object members to be `const`.
+
+== Motivation
+
+Today, the `const` modifier can only be applied to class properties.
+
+However, there are times where this rule is insufficient.
+
+When a module amends another module, it has special rules; top-level properties are considered _object properties_, and not _class properties_.
+
+This means that these properties cannot be declared `const`.
+
+[source,pkl]
+----
+amends "foo.pkl"
+
+const bar = 5 // Pkl Error: modifier `const` cannot be applied to object members
```

Comment:
Here, the `const`-ness is not the issue. Either `bar` is `const` in `foo.pkl`, in which case it couldn't be overridden, or `bar` doesn't occur in `foo.pkl`, so it can't be added here.

The problem description only makes sense in the context of `local`s, in my mind.

Reply:
I think I'll just get rid of this block. You're right; it's not describing what the problem actually is.